internal/metrics/querynode.go

```diff
@@ -0,0 +1,265 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/milvus-io/milvus/internal/util/typeutil"
+)
+
+const (
+	// TODO: use the common status label
+	queryNodeStatusLabel        = "status"
+	QueryNodeMetricLabelSuccess = "success"
+	QueryNodeMetricLabelFail    = "fail"
+	QueryNodeMetricLabelTotal   = "total"
+
+	// TODO: use the common status label
+	nodeIDLabel       = "node_id"
+	collectionIDLabel = "collection_id"
+)
+
+const (
+	// query type
+	queryTypeLabel           = "query_type"
+	QueryNodeQueryTypeSearch = "search"
+	QueryNodeQueryTypeQuery  = "query"
+
+	// segment type
+	segmentTypeLabel        = "segment_type"
+	QueryNodeSegTypeSealed  = "sealed"
+	QueryNodeSegTypeGrowing = "growing"
+)
+
+// queryNodeDurationBuckets involves durations in milliseconds
+var queryNodeDurationBuckets = []float64{0, 10, 100, 1000, 10000, 100000, 1000000, 10000000}
```

Comment:
buckets no need to have a 0 bucket. Because all the buckets has the meaning of less than specified value. 
The the buckets value is 10 times greater than the before. Normally, we always use the power of 2 .

Reply:
Add More buckets